packages/root/src/Root.spec.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { render, cleanup } from '@testing-library/react';
+import Root from '.';
+
+afterAll(cleanup);
+
+describe('packages/root', () => {
+  describe('by default', () => {
+    const props = { name: 'testName' };
+    const content = 'button content';
+    const { getByTestId } = render(
```

Comment:
I've seen this throughout, but I'm uncomfortable with several `render` calls if we're not also taking care of the previously existing elements on the body. In this pattern, each time you call `render` it will be added to the same body, and the order of the tests (and render calls) will impact the results of the test. 

As a result, I prefer to be very strict about using `beforeEach` and `afterEach` similar to Mocha/Chai. That doesn't mean we need to sacrifice what React Testing Library does for us (heavily restricting us to UI actions instead of modifying State/Props/etc). It just means we have to be very deliberate about scope and ensuring every test runs in isolation. 

It can be tempting to do `const { getByTestId, rerender } = render(...)`, but I would push heavily against the use of `rerender` unless you are literally forcing a re-render to account for asynchronous stubbed changes, and jest may do a better job of handling that anyway.

So instead I would look for something like this, in pseudo-code:
```
describe packages/mine
  let <any variables to be referenced repeatedly in tests>;

  beforeEach
    set up any values that will be used throughout the test
    set up your stubs here so they're consistent and in one place

  afterEach
    document.body.innerHTML = "";
    do whatever you need to restore stubs
    cleanup() (clean up each time, rather than only at the very end of the suite like in an afterAll call)

  describe when rendering X component
    describe with Y state
      beforeEach
        render component with props for Y
      test
        assert something about component and state  
      describe and I do an action
        beforeEach
          do the action
        test
          assert that i did the action
    describe with Z state
      beforeEach
        render component with props for Z
      test
        assert something about component and state  
```

Let me know if you have questions about this.

Reply:
Note: I used `document.body.innerHTML` to wipe everything, but it would probably be better to take `unmount` from the call to `render` and set a variable scoped for all of your tests and call `unmount && unmount();` in the root level `afterEach` call.

FWIW `unmount` will definitely trigger any hooks etc that occur at the end of the component lifecycle. What I did *should*, but I am a little nervous now that I think about it.